vendor/nanovg/gl/gl.odin

```diff
@@ -0,0 +1,1452 @@
+package nanovg_gl
+
+import "core:log"
+import "core:strings"
+import "core:mem"
+import "core:math"
+import "core:fmt"
+import gl "vendor:OpenGL"
+import nvg "../../nanovg"
+
+Color :: nvg.Color
+Vertex :: nvg.Vertex
+ImageFlags :: nvg.ImageFlags
+TextureType :: nvg.Texture
+Paint :: nvg.Paint
+ScissorT :: nvg.ScissorT
+
+CreateFlag :: enum {
+	// Flag indicating if geometry based anti-aliasing is used (may not be needed when using MSAA).
+	ANTI_ALIAS,
+	// Flag indicating if strokes should be drawn using stencil buffer. The rendering will be a little
+	// slower, but path overlaps (i.e. self-intersecting or sharp turns) will be drawn just once.
+	STENCIL_STROKES,
+	// additional debug checks
+	DEBUG,
+}
+CreateFlags :: bit_set[CreateFlag]
+
+FRAG_BINDING :: 0
+USE_STATE_FILTER :: true
+GL_UNIFORMARRAY_SIZE :: 11
+
+UniformLoc :: enum {
+	VIEW_SIZE,
+	TEX,
+	FRAG,
+}
+
+ShaderType :: enum i32 {
+	FILL_GRAD,
+	FILL_IMG,
+	SIMPLE,
+	IMG,
+}
+
+Shader :: struct {
+	prog: u32,
+	frag: u32,
+	vert: u32,
+	loc: [UniformLoc]i32,
+}
+
+Texture :: struct {
+	id: int,
+	tex: u32,
+	width, height: int,
+	type: TextureType,
+	flags: ImageFlags,
+}
+
+Blend :: struct {
+	src_RGB: u32,
+	dst_RGB: u32,
+	src_alpha: u32,
+	dst_alpha: u32,
+}
+
+CallType :: enum {
+	NONE,
+	FILL,
+	CONVEX_FILL,
+	STROKE,
+	TRIANGLES,
+}
+
+Call :: struct {
+	type: CallType,
+	image: int,
+	pathOffset: int,
+	pathCount: int,
+	triangleOffset: int,
+	triangleCount: int,
+	uniformOffset: int,
+	blendFunc: Blend,
+}
+
+Path :: struct {
+	fillOffset: int,
+	fillCount: int,
+	strokeOffset: int,
+	strokeCount: int,
+}
+
+when GL2_IMPLEMENTATION {
+	FragUniforms :: struct #raw_union {
+		using _: struct {
+			scissorMat: [12]f32, // matrices are actually 3 vec4s
+			paintMat: [12]f32,
+			innerColor: Color,
+			outerColor: Color,
+			scissorExt: [2]f32,
+			scissorScale: [2]f32,
+			extent: [2]f32,
+			radius: f32,
+			feather: f32,
+			strokeMult: f32,
+			strokeThr: f32,
+			texType: i32,
+			type: ShaderType,
+		},
+		uniform_array: [GL_UNIFORMARRAY_SIZE][4]f32,
+	}
+} else {
+	FragUniforms :: struct #packed {
+		scissorMat: [12]f32, // matrices are actually 3 vec4s
+		paintMat: [12]f32,
+		innerColor: Color,
+		outerColor: Color,
+		scissorExt: [2]f32,
+		scissorScale: [2]f32,
+		extent: [2]f32,
+		radius: f32,
+		feather: f32,
+		strokeMult: f32,
+		strokeThr: f32,
+		texType: i32,
+		type: ShaderType,
+	}
+}
+
+GL2_IMPLEMENTATION :: false
+GL3_IMPLEMENTATION :: true
+GLES2_IMPLEMENTATION :: false
+GLES3_IMPLEMENTATION :: false
```

Comment:
I'd make these definable through `#config`? Is this a good idea as I am not sure myself?
```odin
DEFAULT_IMPLEMENTATION_STRING :: #config(NANOVG_GL_IMPL, "GL3")
GL2_IMPLEMENTATION   :: DEFAULT_IMPLEMENTATION_STRING  == "GL2"
GL3_IMPLEMENTATION   :: DEFAULT_IMPLEMENTATION_STRING  == "GL3"
GLES2_IMPLEMENTATION :: DEFAULT_IMPLEMENTATION_STRING  == "GLES2"
GLES3_IMPLEMENTATION :: DEFAULT_IMPLEMENTATION_STRING  == "GLES3"
```